pandas/tests/test_base.py

```diff
@@ -1051,6 +1051,22 @@ def test_searchsorted(self):
             self.assertTrue(0 <= index <= len(o))
 
 
+class TestUInt64HashTable(tm.TestCase):
+
+    def test_lookup_overflow(self):
+        from pandas.hashtable import UInt64HashTable
+        xs = np.array([1, 2, 2**63], dtype=np.uint64)
+        m = UInt64HashTable()
+        m.map_locations(xs)
+        self.assert_numpy_array_equal(m.lookup(xs),
+                                      np.arange(len(xs), dtype=np.int64))
+
+    def test_get_unique(self):
+        s = pd.Series([1, 2, 2**63, 2**63], dtype=np.uint64)
```

Comment:
Okay, well that's a larger issue beyond this PR.  However, let's not merge until I patch `pd.unique`, which suffers from a similar issue as `Series.unique`.